src/main/docker/scripts/export-file-based-secrets.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+#
+# Copyright 2017-2021 Micro Focus or one of its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# A function for exporting file-based secrets.
+#
+# For example, for each environment variable ending in the _FILE suffix:
+#
+#     ABC_PASSWORD_FILE=/var/somefile.txt
+#
+# read the contents of /var/somefile.txt (for example 'mypassword'), and export an environment variable named ABC_PASSWORD set to:
+#
+#     ABC_PASSWORD=mypassword
+export_file_based_secrets() {
+    env -0 | while IFS='=' read -r -d '' env_var_name env_var_value; do
+        if  [[ ${env_var_name} == *_FILE ]] ;
+        then
+            local env_var_name_without_file_suffix=${env_var_name%_FILE}
+            if [ "${!env_var_name:-}" ] && [ "${!env_var_name_without_file_suffix:-}" ]; then
+                echo "ERROR: Both $env_var_name and $env_var_name_without_file_suffix are set (but are exclusive)"
+                exit 1
+            fi            
+            echo "INFO: Found environment variable ending with the _FILE suffix: $env_var_name=$env_var_value, attempting to read the \
+contents of $env_var_value..."
+            if [ -e "$env_var_value" ]; then
+                local file_contents=$(<${env_var_value})
+                echo "INFO: Successfully read contents of ${env_var_value}, exporting environment variable \
+$env_var_name_without_file_suffix using the contents of ${env_var_value} as the value..."
+                if export "$env_var_name_without_file_suffix"="$file_contents" ; then
```

Comment:
I'm not sure the `export` is working. When I created a derived image:

Dockerfile
```
FROM dev/cafapi/opensuse-base:2.4.0-SNAPSHOT
```

Built it:
```
docker build -t test .
```

And the ran it passing it the `env` command:
```
docker run -e ABC_PASSWORD_FILE=/pw.txt -d test env
[14:40:54.825Z #006.??? INFO  -            -   ] startup.sh: Running export-file-based-secrets.sh...
[14:40:54.831Z #00A.??? INFO  -            -   ] export-file-based-secrets.sh: Found environment variable ending with the _FILE suffix: ABC_PASSWORD_FILE=/pw.txt, attempting to read the contents of /pw.txt...
[14:40:54.834Z #00A.??? INFO  -            -   ] export-file-based-secrets.sh: Successfully read contents of /pw.txt, exporting environment variable ABC_PASSWORD using the contents of /pw.txt as the value...
[14:40:54.837Z #00A.??? INFO  -            -   ] export-file-based-secrets.sh: Successfully exported environment variable ABC_PASSWORD
...
LANG=en_US.utf8
HOSTNAME=403449a86633
TINI_VERSION=v0.18.0
ABC_PASSWORD_FILE=/pw.txt
PWD=/
HOME=/root
SHLVL=0
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

the `ABC_PASSWORD` env var is not there (just the `ABC_PASSWORD_FILE` env var passed to the run command)

I'll need to look into it a bit more.

Reply:
Should be good now, had to rework a couple of things: https://github.com/CAFapi/opensuse-base-image/pull/34/commits/15a66024b22b4df3bc9e434e0bc34e6f80215da5